### DISCUSSION TITLE
Systems Metadata Expansion

### DISCUSSION THREAD

**Discussion by G-Ambatte** [naturalcrit/homebrewery#1212] [Ideas]
Currently Homebrewery has the following options for "system":
- 5e
- 4e
- 3.5e
- Pathfinder

@ericscheid:
> all my stuff is 2e.

@G-Ambatte:
> It looks to be pretty trivial to add another system to the list (metadataEditor.jsx:8 - `const SYSTEMS = ['5e', '4e', '3.5e', '2e', 'Pathfinder'];`). The question then becomes "If we start down this road, when and where do we stop?"  
Checking against DriveThruRPG.com, they list 10 different options for D&D/OGL, Pathfinder adds another 3, then there's GURPS and D100 and Savage Worlds and the list goes on and on forever. I suspect the best way to handle it is to list the three or four major systems/versions and include an "Other" category for everything else. When searching, include the option of "Uncategorized" for brews where no system has been selected.

@ericscheid:
> Yep. I'd be happy with an `OSR` broad category (vs finicky 1e, 2e, 2.5e, BECMI, Moldvay, LBB, etc etc)

---

So the question remains for discussion: what systems should Homebrewery include? How do we organise them so it is easy to include or remove them in the Brew metadata? Where do we draw the line?

**Comment by GatesShawn** [comment]
The WoD/CofD community is asking for support, how likely would it be for that to happen (I'm offering to work on it)?

**Reply by ericscheid** [reply]
The code change itself is very simple, but the real question is how we are intending to handle this metadata, or more exactly how are we going to handle ongoing expansion of the metadata tagging.

Right now each brew could have multiple selections of the `SYSTEMS`, and adding new systems into that is as easy as adding new strings into the array.

Side note: I've also been thinking that perhaps a system-specific stylesheet could be loaded, according to that system selection. Thus, one could say "this brew is a WoD thing" and then the WoD-stylesheet gets loaded. Crazy idea, and not entirely compatible with the "select as many systems as you want" metadata structure we have right now.

So .. what does "support for WoD/CofD" look like to you? Is it simply being able to hit a checkbox like we do for now (which then gets totally ignored), will it simply facilitate finding brews for nominated systems (i.e. a simple search tool for published brews), or more?

**Reply by GatesShawn** [reply]
The people interested would in fact want custom stylesheets. Actually, the various lines within WoD/CofD actually have their own styles each; i.e. Vampire: The Masquerade vs Geist: The Sin-Eaters have different fonts, colors etc.

So just expanding the current meta data tag is the least part of it for supporting that.

**Reply by calculuschild** [reply]
> The people interested would in fact want custom stylesheets. Actually, the various lines within WoD/CofD actually have their own styles each; i.e. Vampire: The Masquerade vs Geist: The Sin-Eaters have different fonts, colors etc.
> 
> So just expanding the current meta data tag is the least part of it for supporting that.

Depending what you want to do, you can simply modify the CSS on the page to change the appearance of your document to match another style. For example, we have a user on Reddit @Gazook who often makes [custom styles](https://www.reddit.com/r/homebrewery/comments/leeju9/theme_add_2e_monstrous_compendium_v1/) to match older versions of D&D. You can see how he approaches it in his documents (and someone on the subreddit might be willing to guide you on making your own as well).

**Reply by GatesShawn** [reply]
I know that some CSS can cover that, but I think the overall community is hoping for something that they can easily use to switch between a bunch of options easily. I think a Fork and rebuild to make it support switching styles for each line may be the best option.

**Comment by ericscheid** [comment]
Time has passed.

Two notable points:

1. considerable progress has been made in building `themes` subsystem for styles.  
(not quite pushed to live as yet, and muc work to be done writing the actual themes too).
3. the `brew.system` metadata field will be going away,  
instead, systems for a brew will be indicated via the `brew.tags` metadata field,  
recorded as `system:5e` and `system:pathfinder` and such forth.  
tags are open-ended, and so users can add new `system:*` tags as they feel the need

**Reply by Gazook89** [reply]
> the brew.system metadata field will be going away,

This is an old discussion but relevant right now, and I am curious how we will eliminate this field.  Should we migrate existing `brew.system` info on brews that have this data into the new tags, automatically? 

@ericscheid @calculuschild @dbolack-ab 

*feel free to discuss in gitter, i just wanted to be sure there was a slightly more permanent record here as well if we decide one or the other*

**Reply by dbolack-ab** [reply]
If we can relatively painlessly do a mass update, I would do so. Otherwise, I would do it passively on document edit.
